module/behaviour/skills/FallingRelax/src/FallingRelax.cpp

```diff
@@ -34,54 +144,63 @@
-            FALLING_ANGLE           = std::cos(fallingAngle);
-
-            // When falling the acceleration should drop below this value
-            FALLING_ACCELERATION = config["FALLING_ACCELERATION"].as<float>();
+            log<NUClear::INFO>("FallingRelax - Config");
+            log_level                 = config["log_level"].as<NUClear::LogLevel>();
+            cfg.falling_angle         = config["falling_angle"].as<float>();
+            cfg.falling_acceleration  = config["falling_acceleration"].as<float>();
+            cfg.recovery_acceleration = config["recovery_acceleration"].as<std::vector<float>>();
+            cfg.relax_fall_priority   = config["relax_fall_priority"].as<float>();
 
-            // Once the acceleration has stabalized, we are no longer falling
-            RECOVERY_ACCELERATION = config["RECOVERY_ACCELERATION"].as<std::vector<float>>();
-
-            PRIORITY = config["PRIORITY"].as<float>();
         });
 
-        on<Last<5, Trigger<Sensors>>, Single>([this](const std::list<std::shared_ptr<const Sensors>>& sensors) {
-            if (!falling && !sensors.empty() && fabs(sensors.back()->Htw(2, 2)) < FALLING_ANGLE) {
+        emit<Scope::INITIALIZE>(std::make_unique<RegisterAction>(RegisterAction{
+            subsumption_id,
+            "Falling Relax",
+            {std::pair<float, std::set<LimbID>>(
+                0,
+                {LimbID::LEFT_LEG, LimbID::RIGHT_LEG, LimbID::LEFT_ARM, LimbID::RIGHT_ARM, LimbID::HEAD})},
+            [this](const std::set<LimbID>& /* limbs */) { emit(std::make_unique<ExecuteFallingRelax>()); },
+            [this](const std::set<LimbID>& /* limbs */) { emit(std::make_unique<KillFallingRelax>()); },
+            [this](const std::set<ServoID>& /* servos */) { emit(std::make_unique<KillFallingRelax>()); }}));
 
-                // We might be falling, check the accelerometer
-                double magnitude = 0;
+        on<Last<5, Trigger<Sensors>>>().then("FallingRelax Fallen Check",
+            [this](const std::list<std::shared_ptr<const Sensors>>& sensors) {
 
-                for (const auto& sensor : sensors) {
-                    magnitude += sensor->accelerometer.norm();
-                }
+            //Transform to torso {t} from world {w} space
+            Eigen::Matrix4d Hwt = sensors.back()->Htw.inverse();
+            // Basis Z vector of torso {t} in world {w} space
+            Eigen::Vector3d uZTw = Hwt.block(0, 2, 3, 1);
 
-                magnitude /= sensors.size();
+            if (!relaxed && !sensors.empty() && acos(Eigen::Vector3d::UnitZ().dot(uZTw)) > cfg.falling_angle) {
+
+                // We might be falling, check the accelerometer
+                double gravity = sensors.back()->accelerometer.norm();
 
-                if (magnitude < FALLING_ACCELERATION) {
-                    falling = true;
-                    updatePriority(PRIORITY);
+                if (gravity < cfg.falling_acceleration) {
```

Comment:
Have we found that on the real robot this is sufficient to detect falling with the amount of noise that occurs on the IMU? If not, I would suggest breaking this out to a function which includes filtering/cutoffs appropriately so that we don't randomly collapse during play.